src/components/List.jsx

```diff
@@ -1,23 +1,54 @@
 import React from 'react';
 import db from '../lib/firebase';
-import { collection } from 'firebase/firestore';
+import { collection, doc, updateDoc } from 'firebase/firestore';
 import { useCollection } from 'react-firebase-hooks/firestore';
+import moment from 'moment';
 
 export const List = ({ token }) => {
   const [value, loading, error] = useCollection(collection(db, token), {
     snapshotListenOptions: { includeMetadataChanges: true },
   });
 
+  const calcTimeDiff = (purchasedTime) => {
+    const timeNow = moment().format();
+    const date1 = moment(timeNow, 'YYYYMMDD HH:mm:ss');
+    const date2 = moment(purchasedTime, 'YYYYMMDD HH:mm:ss');
+    const timeDiff = date1.diff(date2, 'hours');
+
+    return timeDiff < 24;
+  };
+
+  const updateDocument = async (id) => {
+    const docRef = doc(db, token, id);
+    await updateDoc(docRef, {
+      purchased_date: moment().format(),
+    });
+  };
+
+  const handleClick = (data, e) => {
+    updateDocument(data.id);
+  };
   return (
     <div>
       <h4> Shareable List Token : {token} </h4>
       {error && <strong>Error: {JSON.stringify(error)}</strong>}
       {loading && <span>Collection: Loading...</span>}
       Collection
       {value && (
-        <ul>
+        <ul className="collection-list">
           {value.docs.map((doc) => (
-            <li key={doc.id}>{doc.data().item}</li>
+            <li key={doc.id}>
+              <input
+                type="checkbox"
+                checked={calcTimeDiff(doc.data().purchased_date)}
+                disabled={calcTimeDiff(doc.data().purchased_date)}
+                name={doc.id}
+                id={doc.id}
+                onClick={(e) => handleClick(doc, e)}
+                onChange={(e) => handleClick(doc, e)}
+              />
+              <label>{doc.data().item}</label>
```

Comment:
We'd get a nice little accessibility bump by adding `htmlFor`
```suggestion
              <label htmlFor={doc.id}>{doc.data().item}</label>
```